sqlmesh/core/model/definition.py

```diff
@@ -657,15 +657,23 @@ def render_merge_filter(
         return rendered_exprs[0].transform(d.replace_merge_table_aliases)
 
     def render_physical_properties(self, **render_kwargs: t.Any) -> t.Dict[str, exp.Expression]:
-        def _render(expression: exp.Expression) -> exp.Expression:
+        def _render(expression: exp.Expression) -> exp.Expression | None:
             # note: we use the _statement_renderer instead of _create_renderer because it sets model_fqn which
             # in turn makes @this_model available in the evaluation context
-            rendered_exprs = self._statement_renderer(expression).render(**render_kwargs)
-
-            if not rendered_exprs:
-                raise SQLMeshError(
+            render_kwargs["model_kind_name"] = self.kind.name
```

Comment:
We should make this available not just for physical properties, but more generally too. Can we add this where we add `this_model`?

Reply:
Looks like we already do this below. Why do we need this here?